### DISCUSSION TITLE
How to force display of drag & drop  handle with mv-list / mv-list-item ?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#815] [Q&A] [answered]
Hi everyone, hope you're all fine

When using mv-list / mv-list-item in `<ul> <li>` elements:

I would like to know if there's a way to 'force' display of drag & drop handle ?
(and keep drag & drop already implemented function on line) 

you can see use case n°3 here:

[https://codepen.io/GATIGA/pen/oNGRMRW](https://codepen.io/GATIGA/pen/oNGRMRW)

... I'm a little bit confused because Mavo is displaying different  item-bar 's depending on elements...

Regards

**Comment by LeaVerou** [comment]
You mean to always show the handle even when the item is not hovered?

CSS to the rescue!

```css
.mv-item-bar.mv-ui.mv-ui {
	opacity: 1;
}
```

(the doubled class is to increase specificity)

Note that it still won't show up when the app is not in edit mode. To force the app to always be in edit mode, you can use `mv-mode="edit"`.

**Reply by GalinhaLX** [reply]
Thanks @LeaVerou for the reply, but it's not what I want to do

**Reply by GalinhaLX** [reply]
By the way, interesting case because I so guess that all existing interface is always there, and you just change 'opacity' to show it

In my case, I want to show, something which is not yet there :)

**Comment by DmitrySharabin** [comment, marked as answer]
It seems to me that the question might also be whether it’s possible to show drag handles inside _collections of primitives_ (that are optional by default). I also wonder, is it possible? Or the only way to move primitives inside a collection is by dragging? 🤔

I haven't checked yet, but will it help if we add `mv-item-bar="with move"` to the collection item?

**Reply by DmitrySharabin** [reply]
> I haven't checked yet, but will it help if we add mv-item-bar="with move" to the collection item?

Yes, it works! 🥳

**Reply by GalinhaLX** [reply]
Exactly what I want to do!Thank you so much @DmitrySharabin 


Discovering this mv-item-bar "option" with you, 
but I can't see where to find this option in documentation:
is this function only part of next release, or I'm missing something from actual documentation?

Regards

**Reply by LeaVerou** [reply]
> Exactly what I want to do!Thank you so much @DmitrySharabin
> 
> Discovering this mv-item-bar "option" with you, but I can't see where to find this option in documentation: is this function only part of next release, or I'm missing something from actual documentation?
> 
> Regards

Yup, it's part of the upcoming release, so not documented yet. Glad it solved your use case!

**Reply by GalinhaLX** [reply]
I would like to take this opportunity to congratulate the incredible work of all the awesome MAVO [contributors](https://github.com/mavoweb/mavo/graphs/contributors).

In particular, I am very pleasantly surprised by the reactivity of your answers @LeaVerou and @DmitrySharabin , which are always very precise and rich in information

The more I discover, the more I love MAVO !

For now, I contribute with my humble low level, at least by participating in these discussions of questions / answers

**Reply by GalinhaLX** [reply]
In fact it's "almost" what I want @DmitrySharabin :)

Yes, the `mv-item-bar="with move"` forces the display of handle (great!)
but I also removes the natural drag& drop, when trying to grab the text

Is there a way to have both? (so users can use 2 way to drag & drop)

1. natural drag & drop on present on text
2. displayed handle button mon mv-item-bar

[https://codepen.io/GATIGA/pen/vYeqOvE](https://codepen.io/GATIGA/pen/vYeqOvE)

**Reply by LeaVerou** [reply]
> In fact it's "almost" what I want @DmitrySharabin :)
> 
> Yes, the `mv-item-bar="with move"` forces the display of handle (great!) but I also removes the natural drag& drop, when trying to grab the text
> 
> Is there a way to have both? (so users can use 2 way to drag & drop)
> 
> 1. natural drag & drop on present on text
> 2. displayed handle button mon mv-item-bar
> 
> [codepen.io/GATIGA/pen/vYeqOvE](https://codepen.io/GATIGA/pen/vYeqOvE)

Add `mv-drag-handle` to the collection item. 😄

**Reply by GalinhaLX** [reply]
I tried
```
<ul mv-list mv-initial-items="1">
       <li mv-list-item="name3" mv-item-bar="with move"  mv-drag-handle>list li WITH handle  <strong style="color:red;"><- handle is present, but how to keep 'natural' drag & drop on text also?</strong>
       </li>
</ul>
```

doesn't work

am I using it right?

(tried to put it in different places, but still doesn't work...

[https://codepen.io/GATIGA/pen/eYGwpya](https://codepen.io/GATIGA/pen/eYGwpya)

**Reply by DmitrySharabin** [reply]
Lea meant adding class `mv-drag-handle`.

**Reply by GalinhaLX** [reply]
almost there:

class="mv-drag-handle" adds the grab hand on top of text (even on non- editing mode, which becomes strange for UI)

but it doesn't add the real function to drag & drop 

..anything else to add? :)
updated :
[https://codepen.io/GATIGA/pen/eYGwpya](https://codepen.io/GATIGA/pen/eYGwpya)

**Reply by GalinhaLX** [reply]
oups , sorry , it adds it...but with a delay ....OK good 


hourra

**Reply by GalinhaLX** [reply]
ok, I see, as I have a text inside innerHTML, I can not drag when there's text (
in codepen test) it's where I wrote "handle is present, but how to keep 'natural' drag & drop on text also"

the drag function starts...just after, outside it   like inside a padding or margin: good to know
